pkg/service/backup/restore_worker_tables.go

```diff
@@ -209,9 +210,17 @@ func (w *tablesWorker) workFunc(ctx context.Context, run *RestoreRun, target Res
 	// for creating and downloading batches.
 	// Goroutine returns only in case of error or
 	// if the whole table has been restored.
-	return parallel.Run(len(w.hosts), target.Parallel, func(n int) error {
+	return parallel.Run(len(w.hosts), target.Parallel, func(n int) (err error) {
 		// Current goroutine's host
 		h := &w.hosts[n]
+		defer func() {
+			if err != nil {
+				w.metrics.SetRestoreState(run.ClusterID, w.location, target.SnapshotTag, h.Host, metrics.RestoreStateError)
+				return
+			}
+			w.metrics.SetRestoreState(run.ClusterID, w.location, target.SnapshotTag, h.Host, metrics.RestoreStateIdle)
+		}()
```

Comment:
This got me thinking that in current implementation, after encountering an error on given host, we still try to use it for restoring next table. This could potentially mean, that we will be still getting similar errors for this host multiple times. Maybe it would be better to exclude such host from restore process "for good"?

Reply:
Let's leave it as it is. 
Error doesn't mean that there is something wrong with the host.
`error` is returned on multiple different situations.